lib/tests/unit/configuration_test.cc

```diff
@@ -9,56 +7,62 @@
 #define LEATHERMAN_LOGGING_NAMESPACE "puppetlabs.pxp_agent.configuration_test"
 #include <leatherman/logging/logging.hpp>
 
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
 #include <catch.hpp>
 
 #include <string>
 
 namespace PXPAgent {
 
+namespace fs = boost::filesystem;
 namespace HW = HorseWhisperer;
 namespace lth_log = leatherman::logging;
 
-const std::string CONFIG { std::string { PXP_AGENT_ROOT_PATH }
-                           + "/lib/tests/resources/config/empty-pxp-agent.cfg" };
-const std::string TEST_BROKER_WS_URI { "wss:///test_c_t_h_u_n_broker" };
-const std::string CA { getCaPath() };
-const std::string CERT { getCertPath() };
-const std::string KEY { getKeyPath() };
-const std::string MODULES_DIR { std::string { PXP_AGENT_ROOT_PATH }
-                                + "/lib/tests/resources/test_modules/" };
-const std::string SPOOL_DIR { std::string { PXP_AGENT_ROOT_PATH }
-                              + "/lib/tests/resources/test_spool/" };
-
-const char* ARGV[] = { "test-command",
-                       "--config-file", CONFIG.data(),
-                       "--broker-ws-uri", TEST_BROKER_WS_URI.data(),
-                       "--ssl-ca-cert", CA.data(),
-                       "--ssl-cert", CERT.data(),
-                       "--ssl-key", KEY.data(),
-                       "--modules-dir", MODULES_DIR.data(),
-                       "--spool-dir", SPOOL_DIR.data(),
-                       "--foreground=true"};
-const int ARGC = 16;
+static const std::string TEST_BROKER_WS_URI { "wss:///test_c_t_h_u_n_broker" };
+static const std::string CA { TEST_CA_CERT_PATH.string() };
+static const std::string CERT { TEST_CERT_PATH.string() };
+static const std::string KEY { TEST_KEY_PATH.string() };
+static const std::string CONFIG { TEST_EMPTY_CONFIG_PATH.string() };
+static const std::string MODULES_DIR = { TEST_MODULES_DIR_PATH.string() };
+static const std::string SPOOL_DIR { TEST_SPOOL_DIR_PATH.string() };
+
+static const char* ARGV[] = {
+    "test-command",
+    "--config-file", CONFIG.data(),
+    "--broker-ws-uri", TEST_BROKER_WS_URI.data(),
+    "--ssl-ca-cert", CA.data(),
+    "--ssl-cert", CERT.data(),
+    "--ssl-key", KEY.data(),
+    "--modules-dir", MODULES_DIR.data(),
+    "--spool-dir", SPOOL_DIR.data(),
+    "--foreground=true" };
+static const int ARGC = 16;
 
 static void configureTest() {
+    HW::Reset();
+    if (!fs::exists(SPOOL_DIR) && !fs::create_directories(SPOOL_DIR)) {
```

Comment:
Ensure that the testing spool directory is created, since we don't do that anymore.